### DISCUSSION TITLE
wireguard-tools config paths no longer matches paths expected by other tool(s)

### DISCUSSION THREAD

**Discussion by shaynemartin91** [Homebrew/discussions#4430] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.13                                                                     
ORIGIN: https://github.com/Homebrew/brew                                                     
HEAD: 9b60fcf7ac3f82fa589a64091376d0663262f8c6                                               
Last commit: 4 days ago                                                                      
Core tap origin: https://github.com/Homebrew/homebrew-core                                   
Core tap HEAD: 28571ec2168549c6bf5513534c3383759a22d40f                                      
Core tap last commit: 3 days ago                                                             
Core tap branch: master                                                                      
Core tap JSON: 15 Apr 01:41 UTC                                                              
HOMEBREW_PREFIX: /opt/homebrew                                                               
HOMEBREW_CASK_OPTS: []                                                                       
HOMEBREW_EDITOR: /opt/homebrew/bin/hx                                                        
HOMEBREW_MAKE_JOBS: 10                                                                       
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby 
CPU: 10-core 64-bit arm_firestorm_icestorm                                                   
Clang: 14.0.3 build 1403                                                                     
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git                               
Curl: 7.87.0 => /usr/bin/curl                                                                
macOS: 13.3.1-arm64                                                                          
CLT: 14.3.0.0.1.1679647830                                                                   
Xcode: N/A                                                                                   
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers 
with debugging if you file an issue. If everything you use Homebrew for is     
working fine: please don't worry or file an issue; just ignore this. Thanks!   
                                                                               
Warning: Some installed formulae are deprecated or disabled.                   
You should find replacements for the following formulae:                       
  php@7.4                                                                      
  php@8.0
```

### Description of issue

After updating the [wireguard-tools formula](https://formulae.brew.sh/formula/wireguard-tools), my [Pritunl client](https://formulae.brew.sh/cask/pritunl) stopped working. Digging in, I found Pritunl is using `wg-quick` (provided by `wireguard-tools`). Before invoking `wg-quick`, it creates the appropriate config file in `/usr/local/etc/wireguard`. 

Looking at the source, `wg-quick` looks for config files in:
- `/etc/wireguard`
- `/usr/local/etc/wireguard`

I found [this recent commit](https://github.com/Homebrew/homebrew-core/commit/0325ac3a9af666a299feeb66ab0d60e5079419ca) where it looks like the formula was previously replacing `/etc/wireguard` with `#{etc}/wireguard` in the `darwin.bash` file (on my machine that seems to be `/opt/homebrew/etc/wireguard`). But with the change in the commit, it now replaces `/usr/local/etc/wireguard` and can't find the file.

So after the replacement, Pritunl is creating a config in `/usr/local/etc/wireguard` then calling `wg-quick` but `wg-quick` is looking for the config in `/opt/homebrew/etc/wireguard`.

Does this assessment sound correct? If so, I'd love to contribute a fix. But I wanted to check here first since I'm not certain where the right place for the fix is. I see a few ways this could be fixed:

1. Update the `wireguard-tools` formula to append the `#{etc}/wireguard` path instead of replacing
    a. This would make the `wg-quick` tool work for users with either `/usr/local/etc` or `/opt/homebrew/etc`.
2. Update the `pritunl` formula to do a similar replacement
    a. This would only fix Pritunl and only if they install it via homebrew. Other tools might be depending on `wg-quick` using the `/usr/local/etc` directory
3. Update the `pritunl` client source. 
    a.  The client is open source, so it could be updated to use the `/opt/homebrew/etc` directory, but that would break for users that have `wireguard-tools` installed by some method other than homebrew, or with homebrew prefix still set to `/usr/local`

It seems like option 1 is the best approach since it would maintain the widest compatibility. Are there any other approaches that would be better or is there a reason to not allow it to also search the `/usr/local/etc/wireguard` directory?

**Comment by carlocab** [comment]
> 3. Update the `pritunl` client source.
>    a.  The client is open source, so it could be updated to use the `/opt/homebrew/etc` directory, but that would break for users that have `wireguard-tools` installed by some method other than homebrew, or with homebrew prefix still set to `/usr/local`

Would be nice to do something like this, except make it so that `pritunl` also checks `/opt/homebrew/etc` in addition to `/usr/local/etc` so it works for users who keep files in `/opt/homebrew/etc`.

**Comment by ankorez** [comment]
Hello, I recently installed a pritunl server and when I added wireguard on my mac I had the same problem. I came across this topic and it helped me a lot.

I added this path `/usr/local/etc/wireguard` at the file located in`/opt/homebrew/opt/wireguard-tools/bin/wg-quick`

`CONFIG_SEARCH_PATHS=( /etc/wireguard /opt/homebrew/etc/wireguard /usr/local/etc/wireguard )`

the wireguard client works again correctly on my Mac

i've make a command for fix quickly the issue
`sed -i '' 's/CONFIG_SEARCH_PATHS=( \/etc\/wireguard \/opt\/homebrew\/etc\/wireguard )/CONFIG_SEARCH_PATHS=( \/etc\/wireguard \/opt\/homebrew\/etc\/wireguard \/usr\/local\/etc\/wireguard )/' /opt/homebrew/opt/wireguard-tools/bin/wg-quick`
thanks a lot
